weave

```diff
@@ -921,8 +921,8 @@ proxy_args() {
 }
 
 proxy_addr() {
-    if addr=$(docker logs $PROXY_CONTAINER_NAME 2>/dev/null | head -n3 | grep -oE "proxy listening on .*"); then
-      addr=${addr##* }
+    if addr=$(docker logs $PROXY_CONTAINER_NAME 2>/dev/null | grep -oE "proxy listening on .*" | head -n1); then
+      addr=${addr##* tcp://}
```

Comment:
There is no guarantee that the first address appearing in the output is the one we want.
Plus, because there is no separation between the listen and serve phases, we can potentially have lots of output (from the proxy doing its stuff) interspersed with the "proxy listening on ..."